src/datepicker/ngb-date.ts

```diff
@@ -1,6 +1,15 @@
 export class NgbDate {
-  static from(date: {year: number, month: number, day?: number}) {
-    return date ? new NgbDate(date.year, date.month ? date.month : 1, date.day ? date.day : 1) : null;
+  static from(date: {year: number, month?: number, day?: number}) {
+    let nDate = date ? new NgbDate(date.year, date.month ? date.month : 1, date.day ? date.day : 1) : null;
+    if (nDate) {
+      // add 1 to year to prevent any date in last valid javascript Date being picked and causing issues around
+      // September
+      let jDate = new Date(nDate.year + 1, nDate.month - 1, nDate.day);
```

Comment:
I'd argue there aren't any real world use cases that would be redefining dates outside of what can be represented in javascript, and the risk of not implementing it at the `NgbDate` level would be that any other calendar implementation would fall into the same infinite loop trap. If at some point the service was hardened up and wouldn't cause an infinite loop, than it could be safely moved inside the specific calendar.

That being said, I can see where you are coming from, but in that case I think the only solution would be to extend the Calendar interface to include a method to validate a given NgbDate and call that before attaching the NgbDate to the model. Otherwise you could bind an invalid date by typing one in, but not pick it in the popup area, and,  in fact, cause errors generating that calendar view.